common/app/routes/challenges/redux/selectors.js

```diff
@@ -36,14 +36,26 @@ const submitTypes = {
 export const challengeSelector = createSelector(
   state => state.challengesApp.challenge,
   state => state.entities.challenge,
-  (challengeName, challengeMap) => {
-    if (!challengeName || !challengeMap) {
+  state => state.entities.block,
+  (challengeName, challengeMap, blockMap) => {
+    if (
+   !challengeName ||
+   !challengeMap ||
+   !blockMap ||
+   !challengeMap[challengeName] ||
+   !blockMap[challengeMap[challengeName].block]
+) {
       return {};
     }
     const challenge = challengeMap[challengeName];
     const challengeType = challenge && challenge.challengeType;
+    const title = challengeTitleString({
+      blockName: blockMap[challenge.block].title || null,
+      title: challenge.title
+    });
```

Comment:
Since this is the only place this function is used, it does not need to be a util.

Let's simplify here by removing a function and using a simple ternary.

```js
const blockName = blockMap[challenge.block].title;
const title = blockName && title ? `${blockName}: ${challenge.title}` : challenge.title;
```